cmd/config_test.go

```diff
@@ -83,8 +97,239 @@ func TestConfigCmd(t *testing.T) {
+}
+
+func (smh *simpleLogrusHook) Levels() []log.Level {
+	return smh.levels
+}
+func (smh *simpleLogrusHook) Fire(e *log.Entry) error {
+	smh.mutex.Lock()
+	defer smh.mutex.Unlock()
+	smh.messageCache = append(smh.messageCache, *e)
+	return nil
+}
+
+func (smh *simpleLogrusHook) drain() []log.Entry {
+	smh.mutex.Lock()
+	defer smh.mutex.Unlock()
+	res := smh.messageCache
+	smh.messageCache = []log.Entry{}
+	return res
+}
+
+var _ log.Hook = &simpleLogrusHook{}
+
+// A helper funcion for setting arbitrary environment variables and
+// restoring the old ones at the end, usually by deferring the returned callback
+//TODO: remove these hacks when we improve the configuration... we shouldn't
+// have to mess with the global environment at all...
+func setEnv(t *testing.T, newEnv []string) (restoreEnv func()) {
+	actuallSetEnv := func(env []string) {
+		os.Clearenv()
+		for _, e := range env {
+			val := ""
+			pair := strings.SplitN(e, "=", 2)
+			if len(pair) > 1 {
+				val = pair[1]
+			}
+			require.NoError(t, os.Setenv(pair[0], val))
+		}
+	}
+	oldEnv := os.Environ()
+	actuallSetEnv(newEnv)
+
+	return func() {
+		actuallSetEnv(oldEnv)
+	}
+}
+
+var verifyOneIterPerOneVU = func(t *testing.T, c Config) {
+	// No config anywhere should result in a 1 VU with a 1 uninterruptible iteration config
+	sched := c.Execution[lib.DefaultSchedulerName]
+	require.NotEmpty(t, sched)
+	require.IsType(t, scheduler.PerVUIteationsConfig{}, sched)
+	perVuIters, ok := sched.(scheduler.PerVUIteationsConfig)
+	require.True(t, ok)
+	assert.Equal(t, null.NewBool(false, false), perVuIters.Interruptible)
+	assert.Equal(t, null.NewInt(1, false), perVuIters.Iterations)
+	assert.Equal(t, null.NewInt(1, false), perVuIters.VUs)
+	//TODO: verify shortcut options as well?
+}
+
+var verifySharedIters = func(vus, iters int64) func(t *testing.T, c Config) {
+	return func(t *testing.T, c Config) {
+		sched := c.Execution[lib.DefaultSchedulerName]
+		require.NotEmpty(t, sched)
+		require.IsType(t, scheduler.SharedIteationsConfig{}, sched)
+		sharedIterConfig, ok := sched.(scheduler.SharedIteationsConfig)
+		require.True(t, ok)
+		assert.Equal(t, null.NewInt(vus, true), sharedIterConfig.VUs)
+		assert.Equal(t, null.NewInt(iters, true), sharedIterConfig.Iterations)
+		//TODO: verify shortcut options as well?
+	}
+}
+
+var verifyConstantLoopingVUs = func(vus int64, duration time.Duration) func(t *testing.T, c Config) {
+	return func(t *testing.T, c Config) {
+		sched := c.Execution[lib.DefaultSchedulerName]
+		require.NotEmpty(t, sched)
+		require.IsType(t, scheduler.ConstantLoopingVUsConfig{}, sched)
+		clvc, ok := sched.(scheduler.ConstantLoopingVUsConfig)
+		require.True(t, ok)
+		assert.Equal(t, null.NewBool(true, false), clvc.Interruptible)
+		assert.Equal(t, null.NewInt(vus, true), clvc.VUs)
+		assert.Equal(t, types.NullDurationFrom(duration), clvc.Duration)
+		//TODO: verify shortcut options as well?
+	}
+}
+
+func mostFlagSets() []*pflag.FlagSet {
+	// sigh... compromises...
```

Comment:
I would like if we can have todos and explanation why some of the flagsets are not tested

Reply:
Yeah, I may add some of the explanation from the second part of my comment above as a comment in the source here. And there's a bunch of related `TODO`s to this already (`also remove these hacks when we improve the configuration...`, `figure out a better way to handle the CLI flags`), though I can see why the connection not obvious.